Mirror/Weaver/Weaver.cs

```diff
@@ -643,14 +643,7 @@ static MethodDefinition GenerateReadFunction(TypeReference variable)
                     continue;
 
                 // mismatched ldloca/ldloc for struct/class combinations is invalid IL, which causes crash at runtime
-                if (variable.IsValueType)
-                {
-                    worker.Append(worker.Create(OpCodes.Ldloca, 0));
-                }
-                else
-                {
-                    worker.Append(worker.Create(OpCodes.Ldloc, 0));
-                }
+                worker.Append(worker.Create(variable.IsValueType ? OpCodes.Ldloca : OpCodes.Ldloc, 0));
```

Comment:
too many nested things here.  Makes my head spin,   can put the opcode in a local variable?  and then call worker.Create(opcode)

Reply:
Ok